library/core/src/time.rs

```diff
@@ -1698,3 +1728,131 @@ impl Duration {
         )
     }
 }
+
+#[cfg(kani)]
+#[unstable(feature = "kani", issue = "none")]
+pub mod duration_verify {
+    use crate::kani;
+    use super::*;
+
+    impl kani::Arbitrary for Duration {
+        fn any() -> Duration {
+            let secs = kani::any::<u64>();
+            let nanos = kani::any::<u32>();
+            Duration::new(secs, nanos)
+        }
+    }
+
+    #[kani::proof_for_contract(Duration::new)]
+    #[kani::should_panic]
+    fn duration_new() {
+        let secs = kani::any::<u64>();
+        let nanos = kani::any::<u32>();
+        let _ = Duration::new(secs, nanos);
+    }
+    
+    #[kani::proof_for_contract(Duration::from_secs)]
+    fn duration_from_secs() {
+        let secs = kani::any::<u64>();
+        let _ = Duration::from_secs(secs);
+    }
+
+    #[kani::proof_for_contract(Duration::from_millis)]
+    fn duration_from_millis() {
+        let ms = kani::any::<u64>();
+        let _ = Duration::from_millis(ms);
+    }
+
+    #[kani::proof_for_contract(Duration::from_micros)]
+    fn duration_from_micros() {
+        let micros = kani::any::<u64>();
+        let _ = Duration::from_micros(micros);
+    }
+
+    #[kani::proof_for_contract(Duration::from_nanos)]
+    fn duration_from_nanos() {
+        let nanos = kani::any::<u64>();
+        let _ = Duration::from_nanos(nanos);
+    }
+
+    #[kani::proof_for_contract(Duration::as_secs)]
+    #[kani::should_panic]
+    fn duration_as_secs() {
+        let dur = kani::any::<Duration>();
+        let _ = dur.as_secs();
+    }
```

Comment:
In keeping with what we recommend at https://model-checking.github.io/kani/reference/attributes.html#kanishould_panic I'd say it would be great to have a pair of harnesses for each contract: one that either firmly constrains the input to one that will cause a `panic`, or maybe just leave it unconstrained as is; and a second one constrains the input (via assumptions or the likes) to cases where no panic is expected.

Reply:
Cool, just added the matching non-panicking harnesses for all harnesses that use `Duration::any()`. I also added a function `safe_duration` that will always return a `Duration` without panicking.